### ISSUE TITLE
Build fails with Lua 5.4

### ISSUE THREAD

**Issue by nloveladyallen** [CorsixTH/CorsixTH#1723]
```
Scanning dependencies of target RncLib
[  2%] Building CXX object libs/rnc/CMakeFiles/RncLib.dir/rnc.cpp.o
[  5%] Linking CXX static library libRncLib.a
[  5%] Built target RncLib
Scanning dependencies of target CorsixTH_lib
[  8%] Building C object CorsixTH/CMakeFiles/CorsixTH_lib.dir/Src/random.c.o
[ 11%] Building CXX object CorsixTH/CMakeFiles/CorsixTH_lib.dir/Src/bootstrap.cpp.o
In file included from /Users/noah/src/CorsixTH-0.64/CorsixTH/Src/bootstrap.cpp:28:
/Users/noah/src/CorsixTH-0.64/CorsixTH/Src/th_lua.h:112:10: error: no matching function for call to 'lua_resume'
  return lua_resume(L, f, n);
         ^~~~~~~~~~
/usr/local/include/lua/lua.h:300:15: note: candidate function not viable: requires 4 arguments, but 3 were provided
LUA_API int  (lua_resume)     (lua_State *L, lua_State *from, int narg,
              ^
1 error generated.
make[2]: *** [CorsixTH/CMakeFiles/CorsixTH_lib.dir/Src/bootstrap.cpp.o] Error 1
make[1]: *** [CorsixTH/CMakeFiles/CorsixTH_lib.dir/all] Error 2
make: *** [all] Error 2
```

I'm using macOS 10.15.7, with Lua 5.4.0.

**Comment by Alberth289346** [comment]
We are ware of that, Lua 5.4 is not yet supported.

**Comment by TheCycoONE** [comment]
This build failure is already fixed in my WIP 5.4 branch. On one hand I wouldn't mind getting those changes in. On the other - people would just encounter problems much later when they try to load a save.
